Dockerfile-bcoin

```diff
@@ -1,30 +1,59 @@
 FROM nfnty/arch-mini:latest AS base
 
 RUN mkdir -p /code/node_modules/bcoin /data
-WORKDIR /code/node_modules/bcoin
-ENTRYPOINT ["node"]
-CMD ["/code/scripts/docker-bcoin-init.js"]
+WORKDIR /code
 
 RUN pacman -Sy --noconfirm archlinux-keyring && \
-    pacman -Syu --noconfirm nodejs npm && \
+    pacman -Syu --noconfirm nodejs-lts-carbon npm && \
     rm /var/cache/pacman/pkg/*
 
 FROM base AS build
+
 # Install build dependencies
-RUN pacman -Syu --noconfirm base-devel unrar git python2
-#HACK: Node-gyp needs python
-RUN ln -s /usr/bin/python2 /usr/bin/python
+# Note: node-gyp needs python
+RUN pacman -Syu --noconfirm base-devel unrar git python2 \
+    && ln -s /usr/bin/python2 /usr/bin/python
 
-ARG repo=https://github.com/bcoin-org/bcoin.git
 ARG branch=master
-#HACK: Dingle this to trigger rebuilds
+
+# use this to bust the build cache
 ARG rebuild=0
 
 # Install bcoin
-RUN git clone --branch $branch $repo /code/node_modules/bcoin
-RUN npm install --production
+RUN git clone --branch $branch \
+    https://github.com/bcoin-org/bcoin.git /code/node_modules/bcoin \
+    && pushd /code/node_modules/bcoin \
+    && npm install --production \
+    && popd
+
+# Install bmultisig
+RUN git clone --branch $branch \
+    https://github.com/bcoin-org/bmultisig.git /code/node_modules/bmultisig \
+    && pushd /code/node_modules/bmultisig \
+    && npm install --production \
+    && popd
+
+# Install blgr
+RUN git clone --branch $branch \
+    https://github.com/bcoin-org/blgr /code/node_modules/blgr \
+    && pushd /code/node_modules/blgr \
+    && npm install --production \
+    && popd
 
-#TODO: Inherit from official image
+# Install bclient
+RUN git clone --branch $branch \
+    https://github.com/bcoin-org/bclient.git /code/node_modules/bclient \
+    && pushd /code/node_modules/bclient \
+    && npm install --production \
+    && popd
+
+# TODO: Inherit from official image
 FROM base
+
 COPY --from=build /code/node_modules/bcoin /code/node_modules/bcoin
+COPY --from=build /code/node_modules/bmultisig /code/node_modules/bmultisig
+COPY --from=build /code/node_modules/blgr /code/node_modules/blgr
+COPY --from=build /code/node_modules/bclient /code/node_modules/bclient
 COPY ./scripts/ /code/scripts
+
+CMD ["/code/scripts/entrypoint.sh"]
```

Comment:
@Thann wondering what your thoughts are on this pattern? Long term I think it should be nice and give us some good flexibility for the docker image build in being able to edit the script and leave the dockerfile stable

Reply:
It would basically replace lines 5-6 above.

Reply:
I think using "CMD" is appropriate. It's just the default command and can be easily overridden using the "command" feature of docker-compose. 

I think it could *cough* survive being closer to the top as it never changes ^.^

Reply:
I didn't want to put it at the top in this case because `/code/scripts/entrypoint.sh` doesn't exist until the step before it

Reply:
I agree it's a bit weird, but that doesn't really matter because "CMD" just adds metadata saying what the default cmd should be once fully built =]

Reply:
but it also makes sense to keep them close so ppl can easily see where the script comes from, I just figure ppl would be able to figure it out regardless^.^